src/renderer/components/+extensions/attempt-install-by-info/attempt-install-by-info.tsx

```diff
@@ -23,20 +24,34 @@ interface Dependencies {
   attemptInstall: (request: InstallRequest, d: ExtendableDisposer) => Promise<void>;
   getBaseRegistryUrl: () => Promise<string>;
   extensionInstallationStateStore: ExtensionInstallationStateStore;
+  fetch: Fetch;
 }
 
-export const attemptInstallByInfo = ({ attemptInstall, getBaseRegistryUrl, extensionInstallationStateStore }: Dependencies) => async ({
+export const attemptInstallByInfo = ({
+  attemptInstall,
+  getBaseRegistryUrl,
+  extensionInstallationStateStore,
+  fetch,
+}: Dependencies) => async ({
   name,
   version,
   requireConfirmation = false,
 }: ExtensionInfo) => {
   const disposer = extensionInstallationStateStore.startPreInstall();
   const baseUrl = await getBaseRegistryUrl();
   const registryUrl = new URLParse(baseUrl).set("pathname", name).toString();
-  let json: any;
+  let json;
 
   try {
-    json = await downloadJson({ url: registryUrl }).promise;
+    const request = await fetch(registryUrl);
+
+    if (request.status === 404) {
```

Comment:
It would be even better if we check that if the status code is not between 200 and 299